### PR TITLE
drivers/dose: migrate to ztimer_usec

### PR DESCRIPTION
**Pull request by fjmolinas**
### Contribution description

This PR migrate `dose` to use `ztimer`. It's untested.

### Testing procedure

Maybe @benpicco can suggest the best testing procedure?

### Issues/PRs references

Split from #17365

### CODE REVIEW

File: drivers/dose/dose.c [hunk 1]
```diff
@@ -187,9 +185,8 @@ static dose_signal_t state_transit_blocked(dose_t *ctx, dose_signal_t signal)
     (void) signal;
     uint32_t backoff;
 
-    backoff = random_uint32_range(xtimer_usec_from_ticks(xtimer_min_timeout),
-                                  2 * ctx->timeout_base);
-    xtimer_set(&ctx->timeout, backoff);
+    backoff = random_uint32_range(0, 2 * ctx->timeout_base);
```
Comment (benpicco) [inline drivers/dose/dose.c:188]: Just to be sure: What happens when setting a timeout that is below what the `periph_timer` implementation can achieve? Will it just use the smallest timeout instead or can the situation arise that no alarm is generated?

Comment (fjmolinas) [reply drivers/dose/dose.c:188]: For `periph_timer` there is a `min` field in ztimer `ztimer_periph->min`, so if for example calling `ztimer_set(ZTIMER_USEC, 0)`, then the callback will be actually set to `now + ztimer_periph->min`, which should guarantee the callback to trigger. Currently the default `CONFIG_ZTIMER_USEC_MIN` is 10, but this value can be tuned.

**Comment by fjmolinas** [comment]
Do you think you could test this soon @benpicco?

**Comment by benpicco** [comment]
My test setup is rather unreliable and I don't know why, but this is unrelated to this PR.
Anyway, timeouts do work still, so I think this is good.

**Review by benpicco — APPROVED** [review]

**Review by jue89 — APPROVED** [review]
Thank you for the migration!

I had a short test as well. Works well in my local dose network.

**Pull request merged**